packages/dashboard-core-plugins/src/linker/Linker.tsx

```diff
@@ -550,7 +551,7 @@ export class Linker extends Component<LinkerProps, LinkerState> {
       case ToolType.LINKER:
         setDashboardColumnSelectionValidator(
           localDashboardId,
-          this.isColumnSelectionValid
+          this.isColumnSelectionValid as ColumnSelectionValidator
```

Comment:
Instead I think `isColumnSelectionValid` should also take a `Partial<LinkColumn>` rather than just `LinkColumn` - you'll need to put the appropriate checks for undefineds in there as well.

In particular - in LinkerUtils, there's a check on the `columnType`, checking for `null`:
```
    // Null columnType in ending link point allows linking to any type
    const isCompatibleType =
      endColumnType === null ||
      TableUtils.isCompatibleType(startColumnType, endColumnType);
```